src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7020,6 +7062,40 @@ private void ApplyMemberPostConditions(BoundExpression? receiverOpt, MethodSymbo
             ApplyMemberPostConditions(receiverSlot, method);
         }
 
+        // For an instance method, or a non-static local function in an instance method, returns the slot for the `this` parameter
+        // For a static method, or a static local function, or a local function in a static method, returns 0
+        // Otherwise, returns -1
+        private int GetReceiverSlotForMemberPostConditions(MethodSymbol? method)
+        {
+            if (method is null)
+            {
+                return -1;
+            }
+
+            MethodSymbol? current = method;
+            bool anyStatic = current.IsStatic;
+            while (current.ContainingSymbol is MethodSymbol container)
+            {
+                current = container;
+                anyStatic |= container.IsStatic;
```

Comment:
>anyStatic |= container.IsStatic [](http://example.com/codeflow?start=16&length=31)

Consider removing `anyStatic` and change this to `if (container.IsStatic) return 0;`